internal/engine/netx/errorx/errorx.go

```diff
@@ -49,6 +57,36 @@ const (
 	FailureJSONParseError = "json_parse_error"
 )
 
+// TLS alert protocol as defined in RFC8446
+const (
+	// Sender was unable to negotiate an acceptable set of security parameters given the options available.
+	TLSHandshakeFailure = 40
+
+	// Certificate was corrupt, contained signatures that did not verify correctly, etc.
+	TLSBadCertificate = 42
+
+	// Certificate was of an unsupported type.
+	TLSUnsupportedCertificate = 43
+
+	// Certificate was revoked by its signer.
+	TLSCertificateRevoked = 44
+
+	// Certificate has expired or is not currently valid.
+	TLSCertificateExpired = 45
+
+	// Some unspecified issue arose inprocessing the certificate, rendering it unacceptable.
+	TLSCertificateUnknown = 46
+
+	// Certificate was not accepted because the CA certificate could not be located or could not be matched with a known trust anchor.
+	TLSUnknownCA = 48
+
+	// Handshake (not record layer) cryptographic operation failed.
+	TLSDecryptError = 50
```

Comment:
(FWIW, I double-checked the codes with not only qtls but also the RFC and it looks qtls numbers are correct for all the alter codes we have here, _except_ decrypt error, which is `51` also in the RFC.)

Reply:
Oh sorry, yes, that is a typo. 
I didn't incorporate `DecodeError` since we don't have a TLS record layer error type yet, not for QUIC and not for TLS over TCP. `DecryptError` describes a failure during the handshake.

Reply:
Thank you!